adapters/adgeneration/adgeneration.go

```diff
@@ -55,8 +55,13 @@ func (adg *AdgenerationAdapter) MakeRequests(request *openrtb2.BidRequest, reqIn
 	headers := http.Header{}
 	headers.Add("Content-Type", "application/json;charset=utf-8")
 	headers.Add("Accept", "application/json")
-	if request.Device != nil && len(request.Device.UA) > 0 {
-		headers.Add("User-Agent", request.Device.UA)
+	if request.Device != nil {
+		if len(request.Device.UA) > 0 {
+			headers.Add("User-Agent", request.Device.UA)
+		}
```

Comment:
@banakemi Can you please also add/modify existing JSON tests for the changes you've made in this file?

Reply:
@mansinahar 
Thank you. I fixed it.